app/assets/javascripts/organization-filtering.js

```diff
@@ -0,0 +1,36 @@
+$(document).ready(function() {
+  const $searchAndSort = $("#js-search-and-sort-component");
+
+  if (!$searchAndSort) return;
+
+  const debounce = (function() {
+    let timer = 0;
+
+    return function(callback, ms) {
+      if (timer) {
+        clearTimeout(timer);
+      }
+      timer = setTimeout(callback, ms);
+    };
+  })();
+
+  $("#js-filtering-component").on("change keyup input", function(e) {
+    const componentContext = $(e.target).closest("#js-filtering-component");
+    const $searchForm = componentContext.find("#js-filtering-form");
+    const formData = $searchForm.find('input[name!=utf8]').serialize();
```

Comment:
The above two fixes work... but then we get to this line which would crash if we make those changes. The same pattern applies (but I wrap it in $() to get the serialize function):

```suggestion
    const formData = $(document.getElementById('your-classrooms-filter')).serialize();
```

Reply:
This won't quite work. The `your-classrooms-filter` id is only the query part of the form and we want to serialize all of the fields. It looks like we're not using that id anywhere so I've gone ahead and removed it.

If we don't care about including `utf8=%E2%9C%93` in the URL params, we can do `$(searchForm).serialize()`.

If we want to stay consistent with the other forms on Classroom whose params end up in the URL, we should do `$(searchForm).find('input[name!=utf8]').serialize()` to filter out the utf8 param.

For consistency, I've got `$(searchForm).find('input[name!=utf8]').serialize()` right now, but, I'm happy to change it (no strong feelings on this one).

Reply:
Ah, that makes a lot of sense. Let's leave it as you have it! 🙌